#!/usr/bin/env python
PACKAGE = "fw_ctrl"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("aoa_m", double_t, 0, "soft minimum angle of attack [deg]", -5.0, -10.0, 2.0)
gen.add("aoa_p", double_t, 0, "soft maximum angle of attack [deg]", 7.0, 3.0, 10.0)
gen.add("delta_aoa", double_t, 0, "soft angle of attack buffer [deg]", 4.0, 0.1, 10.0)
gen.add("sig_aoa_1", double_t, 0, "minimum soft exponential cost at unit input = 1", 0.001, 0.0001, 0.1)

gen.add("h_offset", double_t, 0, "soft nadir terrain offset [m]", 10.0, 0.0, 100.0)
gen.add("delta_h", double_t, 0, "soft nadir terrain buffer [m]", 10.0, 0.1, 100.0)
gen.add("sig_h_1", double_t, 0, "minimum soft exponential cost at unit input = 1", 0.001, 0.0001, 0.1)

gen.add("r_offset", double_t, 0, "soft radial terrain offset [m]", 10.0, 0.0, 100.0)
gen.add("delta_r0", double_t, 0, "soft radial terrain buffer (at zero relative velocity) [m]", 5.0, 0.1, 100.0)
gen.add("k_r_offset", double_t, 0, "gain on relative velocity term of radial offset", 1.0, 0.0, 5.0)
gen.add("k_delta_r", double_t, 0, "gain on relative velocity term of radial buffer", 1.0, 0.0, 5.0)
gen.add("sig_r_1", double_t, 0, "minimum soft exponential cost at unit input = 1", 0.001, 0.0001, 0.1)

exit(gen.generate(PACKAGE, "fw_ctrl", "soft_constraints"))
